packages/@uppy/compressor/src/index.ts

```diff
@@ -1,14 +1,34 @@
-import { BasePlugin } from '@uppy/core'
+import { BasePlugin, Uppy } from '@uppy/core'
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-ignore
 import { RateLimitedQueue } from '@uppy/utils/lib/RateLimitedQueue'
 import getFileNameAndExtension from '@uppy/utils/lib/getFileNameAndExtension'
 import prettierBytes from '@transloadit/prettier-bytes'
 import CompressorJS from 'compressorjs'
-import locale from './locale.js'
 
-export default class Compressor extends BasePlugin {
+import type { Body, Meta, UppyFile } from '@uppy/utils/lib/UppyFile'
+import type { PluginOpts } from '@uppy/core/lib/BasePlugin.ts'
+
+import locale from './locale.ts'
+
+declare module '@uppy/core' {
+  export interface UppyEventMap<M extends Meta, B extends Body> {
+    'compressor:complete': (file: UppyFile<M, B>[]) => void
+  }
+}
+
+interface Options extends PluginOpts, CompressorJS.Options {
```

Comment:
We want export ops for users and our naming pattern has been plugin name + `Opts`.

```suggestion
export interface CompressorOpts extends PluginOpts, CompressorJS.Options {
```